### DISCUSSION TITLE
Component corrupt after Browser back button on render with iOS / Safari

### DISCUSSION THREAD

**Discussion by LTKort** [livewire/livewire#5160] [7. Bugs (archived)]
I have a route (/products/{slugs?}) that renders a livewire component with different slugs etc. The main functionality is filtering.

Everything works fine, but as soon as you click a product, redirect and click the back button (from the browser) and after click another filter; I get a corrupt data error.

This only happens on IOS / Safari.  

What's the best way to debug this? Everything before and after hydration seem normal. It works on Mac Chrome, but why not on IOS chrome?

**Comment by joshhanley** [comment]
@LTKort are you able to replicate the issue consistently in a local environment?

If so, you will need to investigate the payload that generates the checksum and the payload of the next request that is triggering the corruption and compare the two to see what the difference is.

The checksum gets generated here

https://github.com/livewire/livewire/blob/63ad79f9bbad51028e1aaeca774f2f5e22f6578d/src/ComponentChecksumManager.php#L14-L20

I created a tool to help debugging this a while back, which might help, but it is only experimental, so don't run it in production https://github.com/joshhanley/livewire-checksum-logger

Hope this helps!

**Reply by LTKort** [reply]
Thanks for the help @joshhanley 

I can reproduce the issue consistently locally on my Mac with Safari.

I tried to fix it with the PR #4942
Which seems to have some effect on Safari, but not on IOS.

So I will check it out and I'll let you know the results.

**Reply by joshhanley** [reply]
@LTKort no worries! 🙂 Ok, that'd be great, thanks!

**Reply by LTKort** [reply]
@joshhanley I can't find any differences in the data of the checksums. I did change the config to cache the back button, which works, but it does not feel as a permanent solution.

Is there an easy way to compare the request & response data? Or should I, for example, store it in the cache and check the diff?

**Reply by joshhanley** [reply]
@LTKort as in you changed `back_button_cache` config option to `true`? That is fine, but we had some issues with back showing stale data, so it just depends on if that is suitable for your app or not.

But yeah, no easy way, I just store it in a log (I actually use Ray to output it too) and them manually go through and compare.

**Reply by LTKort** [reply]
@joshhanley Yes, i did change the config `back_button_cache` to `true`. But I don't know if it will result in other weird behaviour on other components.

But I tried something else (with `back_button_cache` on `false`). I created a diff tool to check the differences. 
It stores the request / response data in the cache. And on an another view I can compare them. 
On Chrome there were no differences and no errors. But in the same situation on Safari, I got an error the only difference was the `htmlHash`
![Screen Shot 2022-08-11 at 17 17 48](https://user-images.githubusercontent.com/2412670/184169187-c4c5c324-9edb-4432-b733-55220f35c69b.png)

**Reply by joshhanley** [reply]
@LTKort ah interesting! That's odd. I wonder why it works on Chrome but is different on Safari...
